pkg/events/sorting/sorting_test.go

```diff
@@ -297,13 +299,19 @@ func retrieveEventsFromSorter(expectedEventsAmount int, sorterOutputChan chan *t
 	}
 }
 
-func sendTestEvents(sorterInputChannel chan *trace.Event, eventPool []eventsIteration) {
-	startTime := int(time.Now().UnixNano())
-	for _, eventsIteration := range eventPool {
-		time.Sleep(eventsIteration.delay)
-		for _, event := range eventsIteration.events {
-			event.Timestamp = startTime + (event.Timestamp * int(time.Millisecond.Nanoseconds()))
-			sorterInputChannel <- &event
+func sendTestEvents(eventPool []eventsIteration) <-chan *trace.Event {
```

Comment:
Make sendTestEvents the owner of the events channel. This is a common pattern to reason about Go concurrent programs. 
Beyond that, close the channel explicitly and use non buffered channel it's easier to catch data races.